compiler/codegen/OMRCodeGenPhase.cpp

```diff
@@ -439,7 +439,7 @@ OMR::CodeGenPhase::performSetupForInstructionSelectionPhase(TR::CodeGenerator *
    {
    TR::Compilation *comp = cg->comp();
 
-   if (TR::Compiler->target.cpu.isZ() && TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
+   if (TR::Compiler->target.cpu.isZ() && TR::Compiler->om.readBarrierType() != gc_modron_readbar_none)
```

Comment:
@NigelYiboYu FYI if this conflicts with your change in #3680. Similarly for other Z related changes.